### ISSUE TITLE
Question: How to link arguments where target is a list member

### ISSUE THREAD

**Issue by noamsgl** [omni-us/jsonargparse#394]
I'm using PyTorch Lightning and LightningCLI. I have a .yaml specifying a list of trainer callbacks:
```
# trainer.yaml

callbacks:
- class_path: EarlyStopping
  init_args:
    monitor: validation/auroc
    mode: "max"
- class_path: ModelCheckpoint
  init_args:
    monitor: validation/auroclink_arguments
    mode: "max"
    save_top_k: 2
    save_last: true
    dirpath: <should be linked argument>
```

In my use case, the `dirpath` argument to `ModelCheckpoint` needs to be generated from some source variable, resulting in a URI of the form `s3://bucket/folder/run_id/`

However, the `dirpath` argument belongs to `ModelCheckpoint` which is an element of a list. How can I link to this variable using the `jsonargparse.ArgumentLinking.link_arguments` function?

Thanks

**Comment by mauvilsa** [comment]
> the dirpath argument to ModelCheckpoint needs to be generated from some source variable

Which exactly is the source?

**Comment by mauvilsa** [comment]
This has been implemented in pull request #399. The link would be created with target='trainer.callbacks.init_args.dirpath', i.e. the same as if were not a list. This means that if a type accepts both a single subclass or a list of subclasses, the same link applies.
